app/src/main/java/nodomain/freeyourgadget/gadgetbridge/service/devices/hplus/HPlusHandlerThread.java

```diff
@@ -363,13 +363,14 @@ public boolean processRealtimeStats(byte[] data) {
         //Skip when measuring heart rate
         //Calories and Distance are updated and these values will be lost.
         //Because a message with a valid Heart Rate will be provided, this loss very limited
-        if(record.heartRate == ActivityKind.TYPE_NOT_MEASURED) {
+        getDevice().setFirmwareVersion2(null);
+      /*  if(record.heartRate == ActivityKind.TYPE_NOT_MEASURED) {
```

Comment:
Not really but to get a clearer idea of ​​what I did, I preferred to move the whole logic of information to hplussupport